static/app/utils/marked.tsx

```diff
@@ -51,10 +51,28 @@ marked.setOptions({
   renderer: new SafeRenderer(),
   sanitize: true,
 
-  highlight: (code, lang) => {
-    if (Prism.languages[lang]) {
+  highlight: (code, lang, callback) => {
+    if (!lang) {
+      return code;
+    }
+
+    if (lang in Prism.languages) {
       return Prism.highlight(code, Prism.languages[lang], lang);
     }
+
+    import(`prismjs/components/prism-${lang}.min`)
+      .then(() => {
+        callback?.(null, Prism.highlight(code, Prism.languages[lang], lang));
+      })
+      .catch(error => {
+        callback?.(error, code);
+        // eslint-disable-next-line no-console
+        console.warn(
+          'Prism language not found. Error encountered while trying to highlight markdown code block with marked.',
+          error.message
+        );
+      });
```

Comment:
Should we go ahead and put this in a util?

Reply:
Wouldn't we still have to execute `callback?.(null, Prism.highlight(code, Prism.languages[lang], lang));` with a util? (maybe passed in as a callback once the file is loaded)

Reply:
yeah exactly

Reply:
boom https://github.com/getsentry/sentry/pull/44146/commits/f5819e86abc734fe178cc9df7ed76972bba7f235